### PR TITLE
Fix too large click area for extra small button

### LINKED ISSUE #3102: [BUG] XS button click area sometimes overlap with other interactive elements
- [x] I have written a [good issue](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Issue) 

## Describe the bug
The extra small button will in some cases (icon only, mostly) have a much larger hidden click area than the actual button size. This is a problem when buttons are placed besides each other horizontally, as is the case for the new slides design.

This means that I actually click the left button even though I am hovering over the right button (when the button to the right is disabled, in this case)

https://github.com/kirbydesign/designsystem/assets/42470636/d16bc1d4-6116-4ac9-93d2-12581d7cbe73

## Describe how to reproduce the bug
1. Go to https://cookbook.kirby.design/#/home/showcase/slides
2. Click on the nav buttons in a similar fashion as on the linked video

## Which Kirby version was used?
8.6.0 _**WITH**_ the new slides design 

## What was the expected behavior?
Click area should not overlap.

<hr />

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#the-process-of-contributing) correctly.

### Verification
To make sure the bug is not intended behaviour; it should be verified by a member of team Kirby before moving on to implementation. 

- [ ] Make sure the `NOT verified` label has been removed by a member of [team Kirby](https://github.com/kirbydesign/designsystem/blob/main/.github/SUPPORT.md#team-kirby); do not proceed until this is done

### Implementation 
The contributor who wants to implement this issue should: 

- [ ] Make sure you have read: "[Before you get coding](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#before-you-get-coding)".
- [ ] Signal to others you are working on the issue by assigning yourself.
- [ ] Create a branch from the [develop branch](https://github.com/kirbydesign/designsystem/tree/develop) following our [branch naming convention](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Branch). 
- [ ] Create a test that reproduces the bug following guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)". 
- [ ] Write code to fix the bug

### Review
Once the issue has been implemented and is ready for review:

- [ ] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review). 
- [ ] Create a pull-request.

### PR DESCRIPTION
**Pull request by RasmusKjeldgaard**
## Which issue does this PR close?

This PR closes #3102

## What is the new behavior?
The invisible click area around all of our buttons, that make sure they are at least 44x44 px wide, is too big for our extra small button. Sometimes the click area of a button will overlap with other interactive elements, and result in users unexpectedly clicking the button. 

The click area around the extra small button is reduced, which unfortunately makes it less accessible, but the alternative is not feasible. 

In addition to fixing the invisible click area, I have also removed the content layer padding that all our buttons have when they are icon only, as this was also giving similar issues: 

image

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#the-process-of-contributing) correctly.

### Reminders
- [ ] Make sure you have implemented tests following the guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)".
- [ ] Make sure you have updated the cookbook with examples and showcases (for bug fixes, enhancements & new components).

### Review  
- [ ] Determine if your changes are a fix, feature or breaking-change, and add the matching label to your PR. If it is tooling, dependency updates or similar, add ignore-for-release.
- [ ] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review).
- [ ] Request that the changes are code-reviewed 
- [ ] Request that the changes are [UX reviewed](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#ux-review) (only necessary if your PR introduces visual changes)

When the pull request has been approved it will be merged to `develop` by Team Kirby.

### CODE REVIEW

File: libs/designsystem/button/src/button.component.scss [hunk 1]
```diff
@@ -10,14 +10,24 @@ $button-width: (
...
+  * interactive elements. The maximum target size we can safely do is the button's 
+  * minimum width plus its margin on each side
+  */
+  $button-xs-icon-only-size: utils.size('m');
+  $button-xs-click-area-size: $button-xs-icon-only-size + $button-margin * 2;
+  @include utils.accessible-target-size($width: $button-xs-click-area-size);
+
   font-size: utils.font-size('xs');
-  height: utils.size('m');
+  height: $button-xs-icon-only-size;
```
Comment (jakobe) [inline libs/designsystem/button/src/button.component.scss:30]: Is this also a problem vertically? 🤔

File: libs/designsystem/button/src/button.component.scss [hunk 2]
```diff
@@ -10,14 +10,24 @@ $button-width: (
...
 
 @function button-width($key) {
   @return map.get($button-width, $key);
 }
 
 @mixin button-xs {
+  /*
+  * Extra small button needs a smaller click area to e.g. not overlap with other 
+  * interactive elements. The maximum target size we can safely do is the button's 
+  * minimum width plus its margin on each side
```
Comment (jakobe) [inline libs/designsystem/button/src/button.component.scss:23]: Consider [grammar](https://www.hamilton.edu/academics/centers/writing/seven-sins-of-writing/4) 😜 
```suggestion
  * minimum width plus it's margin on each side
```

Comment (RasmusKjeldgaard) [reply libs/designsystem/button/src/button.component.scss:23]: According to your link _its_ seems to be the right choice here:

> It’s is a contraction for “it is.” It’s is never a possessive.
> Its is the possessive for “it.”

File: libs/designsystem/button/src/button.component.scss (hunk 1 above)
Comment (RasmusKjeldgaard) [reply libs/designsystem/button/src/button.component.scss:30]: Yes, we want the click area to be as big as possible both vertically and horizontally to get as close to the 44 x 44 pixels WCAG guideline as we can. 

In this case though, I am just extracting the value to a variable so I can reuse it for both height, width and the calculation for the accessible target size parameter.

File: libs/designsystem/button/src/button.component.scss (hunk 2 above)
Comment (jakobe) [reply libs/designsystem/button/src/button.component.scss:23]: Damn, you're good!!! giphub

File: libs/designsystem/button/src/button.component.scss (hunk 1 above)
Comment (jakobe) [reply libs/designsystem/button/src/button.component.scss:30]: Yeah, but that's actually my point: shouldn't we keep the 44px height? As buttons usually are next to each other but rarely on top of each outer...
`size('m')` = 24px...

**Review by jakobe — APPROVED** [review]
👏🏻

**Pull request merged**